suprb/optimizer/rule/ns/archive.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABCMeta, abstractmethod
+
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from suprb.utils import RandomState
+
+
+class Archive(BaseComponent, metaclass=ABCMeta):
+    """Base Archive Class to store the rules from previous populations"""
+
+    def __init__(self):
+        self.archive = []
+
+    def set_archive(self, pool: list[Rule]):
+        self.archive = pool
+
+    def set_random_state(self, random_state: RandomState):
+        self.random_state_ = random_state
+
+    def get_archive(self):
+        return self.archive
+
+    def extend_archive(self, rules: list[Rule], n: int):
+        self._add_rules_to_archive(rules, n)
+
+    def _add_rules_to_archive(self, rules: list[Rule], n: int):
```

Comment:
abstractmethod?

Reply:
Depends. Then we would need to add `_add_rules_to_archive` to `ArchiveNone` as well (where it does nothing). Up to you

Reply:
Actually I changed it to an abstract method. Makes more sense this way